crates/snp_validation/src/config.rs

```diff
@@ -0,0 +1,177 @@
+
+    /// Prohibited values for `id_key_digest` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    pub id_key_digest_blacklist: Vec<Digest>,
+
+    /// Minimum value for `policy.abi_major`.`policy.abi_minor`
+    #[serde(default)]
+    pub minimum_abi: VersionReq,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_policy_string")]
+    pub policy_flags: Option<u8>,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_platform_string")]
+    pub platform_info_flags: Option<PlatformInfoFlags>,
+}
+
+fn from_policy_string<'de, D>(deserializer: D) -> Result<Option<u8>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    use serde::de::Error;
+
+    let s: &str = Deserialize::deserialize(deserializer)?;
+
+    let mut flags = FlagSet::<PolicyFlags>::from(PolicyFlags::Reserved);
+
+    for flag in s.to_string().split('|') {
+        match flag.trim() {
+            "Debug" => {
+                flags |= PolicyFlags::Debug;
+            }
+            "MigrateMA" => {
+                return Err(D::Error::custom("SNP migration not supported"));
+            }
+            "SingleSocket" => {
+                flags |= PolicyFlags::SingleSocket;
+            }
+            "SMT" => {
+                flags |= PolicyFlags::SMT;
+            }
+            _ => return Err(D::Error::custom(format!("unknown policy flag '{}'", flag))),
+        }
+    }
+
+    Ok(Some(flags.bits()))
+}
+
+fn from_platform_string<'de, D>(deserializer: D) -> Result<Option<PlatformInfoFlags>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    use serde::de::Error;
+
+    let s: &str = Deserialize::deserialize(deserializer)?;
+
+    let flag = match s.trim() {
+        "SME" => PlatformInfoFlags::SME,
+        "TSME" => PlatformInfoFlags::TSME,
+        _ => {
+            return Err(D::Error::custom(format!(
+                "unknown platform info flag '{}'",
+                s
+            )))
+        }
+    };
+
+    Ok(Some(flag))
+}
+
+#[derive(Clone, Debug, Default, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    #[serde(flatten)]
+    pub config: ConfigIntermediate,
+}
+
+impl<'de> Deserialize<'de> for Config {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        use serde::de::Error;
+
+        let config = ConfigIntermediate::deserialize(deserializer)?;
+
+        if config.enarx_signer.is_empty() && config.enarx_hash.is_empty() {
+            return Err(D::Error::custom(
+                "enarx_signer and enarx_hash cannot both be empty",
+            ));
+        }
+
+        Ok(Self { config })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::config::Config;
+    use crate::PolicyFlags;
+    use flagset::FlagSet;
+
+    #[test]
+    fn test_empty_config() {
+        let result: Result<Config, toml::de::Error> = toml::from_str("");
+        if result.is_ok() {
+            unreachable!();
+        }
```

Comment:
no, this is a failed assertion and is very much reachable in your test when implementation is broken. Use `assert!`. Please keep in mind that you or someone else might need to debug this when it breaks. It should be immediately clear what broke if a test fails, `unreachable!` is neither appropriate, not helpful

Reply:
Fixed